structuralcodes/sections/_generic.py

```diff
@@ -0,0 +1,461 @@
+        # I check if point is a shapely Point or an ArrayLike object
+        if not isinstance(point, Point):
+            # It is an ArrayLike object -> create the Point given the
+            # coordinates (coordinates can be a List, Tuple, np.array, ...)
+            coords = np.atleast_1d(point)
+            num = len(coords)
+            if num < 2:
+                raise ValueError('Two coordinates are needed')
+            if num > 2:
+                warn_str = f'Two coordinates are needed. {num}'
+                warn_str += ' coords provided. The extra entries will be'
+                warn_str += ' discarded'
+                warnings.warn(warn_str)
+            point = Point(coords)
+        if not isinstance(material, Material):
+            raise TypeError('The material should be of Material class')
+        self._point = point
+        self._diameter = diameter
+        self._material = material
+        self._area = np.pi * diameter**2 / 4.0
+
+    @property
+    def diameter(self) -> float:
+        '''Returns the point diameter'''
+        return self._diameter
+
+    @property
+    def area(self) -> float:
+        '''Returns the point area'''
+        return self._area
+
+    @property
+    def material(self) -> Material:
+        '''Returns the point material'''
+        return self._material
+
+    @property
+    def x(self) -> float:
+        '''Returns the x coordinate of the point'''
+        return self._point.x
+
+    @property
+    def y(self) -> float:
+        '''Returns the y coordinate of the point'''
+        return self._point.y
+
+    @property
+    def point(self) -> Point:
+        '''Returns the shapely Point object'''
+        return self._point
+
+
+def create_line_point_angle(
+    point: t.Union[Point, t.Tuple[float, float]],
+    theta: float,
+    bbox: t.Tuple[float, float, float, float]
+) -> LineString:
+    '''Creates a line from point and angle within the bounding
+    box'''
+    # create a unit vector defining the line
+    v = (np.cos(theta), np.sin(theta))
+
+    # check if the line is vertical to avoid div by zero
+    if abs(v[0]) > 1e-8:
+        # it is a non vertical line
+        tg = v[1] / v[0]
+        x1 = bbox[0] - 1e-3
+        x2 = bbox[2] + 1e-3
+        y1 = point[1] + (x1 - point[0]) * tg
+        y2 = point[1] + (x2 - point[0]) * tg
+    else:
+        # it is a near-vertical line
+        # atg is almost zero
+        atg = v[0] / v[1]
+        y1 = bbox[1] - 1e-3
+        y2 = bbox[2] + 1e-3
+        x1 = point[0] + (y1 - point[1]) * atg
+        x2 = point[0] + (y2 - point[1]) * atg
+    # create the splitting line
+    return LineString([(x1, y1), (x2, y2)])
+
+
+class SurfaceGeometry:
+    '''Class for a surface geometry with material.
+
+    Basically it is a wrapper for shapely polygon
+    including the material (and other parameters needed)
+    As a shapely polygon it can contain one or more holes'''
+
+    def __init__(
+        self,
+        poly: Polygon,
```

Comment:
Instead of having a `from_points()` method, we could perhaps have

```python
...
poly: t.Union[Polygon, t.Tuple[Point], t.Tuple[t.Tuple[float]]],
...
```

to be able to initialize the `SurfaceGeometry` in a similar manner as the `Polygon` class?